airflow/sensors/time_delta.py

```diff
@@ -80,12 +79,8 @@ def execute(self, context: Context) -> bool | NoReturn:
         if timezone.utcnow() > target_dttm:
             # If the target datetime is in the past, return immediately
             return True
-        try:
-            trigger = DateTimeTrigger(moment=target_dttm, end_from_trigger=self.end_from_trigger)
-        except (TypeError, ValueError) as e:
-            if self.soft_fail:
-                raise AirflowSkipException("Skipping due to soft_fail is set to True.") from e
-            raise
+
+        trigger = DateTimeTrigger(moment=target_dttm, end_from_trigger=self.end_from_trigger)
```

Comment:
wouldn't this cause breaking change in Airflow 2?
https://github.com/apache/airflow/pull/41564 is meant to extract main operators into standard provider. This provider must be supported on Airflow 2 without breaking changes

Reply:
I think we need to merge the provider PR before this one.

Honestly, PRs that handle both core and providers together is really hard to navigate. We should avoid this. It just creates confusion

Reply:
my bad I did not saw that `TimeDeltaSensorAsync` was not using `poke` but directly `execute` ( do you know why ? ) 

so yes this operator is not using the soft_fail logic of the BaseSensorOperator

I just did a revert PR -> https://github.com/apache/airflow/pull/41741

Reply:
Thank you
We can redo the removal on main branch after we release the provider